Capabilities

emaldona opened the pull request:
Related to: #693

edewata commented on src/test/java/org/mozilla/jss/tests/CapabilitiesList.java:
```diff
@@ -0,0 +1,260 @@
+            Provider ps[] = Security.getProviders();
+            lister.listBrief(ps);
+            lister.listVerbose(ps);
+        } catch (Exception e) {
+            Capabilities.logger.info("Exception caught in main: " + e.getMessage(), e);
+        }
```
I'd suggest removing this `try-catch` block too so if something fails the program will return a non-zero code.

edewata commented on src/test/java/org/mozilla/jss/tests/CapabilitiesList.java:
```diff
@@ -0,0 +1,260 @@
+
+        /* Inner class to use existing system nss database
+         */
+        public static class UseSystemDB {
```
If possible avoid using deeply nested classes. Could `UseSystemDB` be moved under `CapabilitiesList`?

edewata reviewed (commented):
Hi @emaldona, I have some comments below, but I don't see any major issues. Please feel free to address them as much as you can, then maybe squash the commits before merging. Thanks!

cipherboy commented on samples/CapabilitiesList/README.md:
```diff
@@ -0,0 +1,50 @@
+Overview
+========================================
+
+**CapabilitiesList** has a Java application to list the capabilities
```
Just a FYI, this directory is going away in a future update :-)

emaldona commented on docs/usage/capabilities_list.md:
```diff
@@ -0,0 +1,75 @@
+attempting to be a default crypto provider, it would be nice to ensure we're 
+mostly close to what the JDK provides. Hopefully other packages written to the
+JDK provider can have similar algorithms available under JSS and behave well. 
+Large gaps in coverage are likely an issue (such as #341 and #242).
```
What do you mean by `linking these issues so it's clickable`

edewata replied:
You can create a link in Markdown like this:
```
such as [#341](https://github.com/dogtagpki/jss/issues/341) ...
```
It will appear like this:
> such as [#341](https://github.com/dogtagpki/jss/issues/341) ...

so you can click the link to open the issue.

emaldona replied:
I do understand now: links to the two issues mentioned in line 16

edewata replied:
Thanks for the update, but it looks like the second link is not rendered correctly.

edewata replied:
@emaldona Here's the first remaining issue.

emaldona replied:
[#341](https://github.com/dogtagpki/jss/issues/341) is okay and the
[#342] (https://github.com/dogtagpki/jss/issues/342) isn't due the extra space between the closing bracket and the opening parenthesis so changing it
[#342](https://github.com/dogtagpki/jss/issues/342) should fix it

emaldona replied:
Not what I intended to type but I bet you understand

edewata commented on docs/usage/capabilities_list.md:
```diff
@@ -0,0 +1,75 @@
+        └── Capabilities4SUN.txt
+
+Here is a brief clip of Capabilities4Mozilla-JSS.txt
+	 AlgorithmParameters : IvAlgorithmParameters
+	 AlgorithmParameters : RC2AlgorithmParameters
+	 AlgorithmParameters : RSAPSSAlgorithmParameters
+	 Cipher : AES
+	 Cipher : DES
+	 Cipher : DESede
+		 Alias: Cipher.DES3
+	 Cipher : RC2
+	 Cipher : RC4
+	 Cipher : RSA
+	 KeyFactory : DSA
+	 KeyFactory : EC
+	 KeyFactory : RSA
+	 KeyGenerator : AES
+	 KeyGenerator : DES
+	 KeyGenerator : DESede
+		 Alias: KeyGenerator.DES3
+	 KeyGenerator : HmacSHA1
+		... ommitted ...
+	 KeyGenerator : HmacSHA512
+	 KeyGenerator : KbkdfCounter
+		 Alias: KeyGenerator.SP800-108-Counter
+		... ommitted ...
```
I meant add a blank line in line 50.

emaldona commented:
What I have done so far in  patch form so I get some preliminary feed back
[possibleFix2ReviewIssues.patch.zip](https://github.com/dogtagpki/jss/files/6518816/possibleFix2ReviewIssues.patch.zip)
Lines 9 though 13 in particular I am not sure about.

emaldona commented on samples/CapabilitiesList/README.md:
```diff
@@ -0,0 +1,50 @@
+
+To install these dependencies on Fedora, execute the following:
+
+    sudo dnf install apache-commons-codec apache-commons-lang gcc-c++ \
+                     java-devel jpackage-utils slf4j zlib-devel \
+                     glassfish-jaxb-api nss-tools nss-devel cmake \
+                     junit
```
Yes, that's better I'll make the change but this file is going away in a future pull request to remove the samples/CapabilitiesList directory. I thing Alex mentioned something like that elsewhere.

edewata replied:
Are you planning to completely remove this file later, or move it somewhere else? If you're removing it anyway that probably should be done in this PR.

emaldona commented on samples/CapabilitiesList/README.md:
```diff
@@ -0,0 +1,50 @@
+see the adjustements4openSUSE.txt file in this directory.
+
+TODO:
+#################################################
```
Thanks, file meant to be removed as stated elsewhere.

emaldona commented on samples/CapabilitiesList/README.md:
```diff
@@ -0,0 +1,50 @@
+Currently doesn't work.
+
+Using make:
+This currently works on Fedora 30 and Debian 10 and also on
```
It does but the file is going away.

edewata commented:
@emaldona Hi, could you apply that patch into your `capabilities` branch and push it to your fork? That way we can actually see the changes in this PR.

edewata commented on docs/usage/capabilities_list.md:
```diff
@@ -0,0 +1,75 @@
+
+Usage
+========================================
+First build jss according to the instructions in jss/README.md
```
It looks like you're linking to the README.md in the master branch. Could you use a relative path?
```
[README](../../README.md)
```

edewata replied:
@emaldona Here's the second remaining issue.

emaldona replied:
That's clear to me.

edewata commented on docs/usage/capabilities_list.md:
```diff
@@ -0,0 +1,78 @@
+contains the application along with the regular tests.
+From the `jss/build` directory execute
+
+    `./run_test.sh org.mozilla.jss.tests.CapabilitiesList`
```
You actually don't need back quotes here since the indentation will automatically make it fixed width.

edewata commented on docs/usage/capabilities_list.md:
```diff
@@ -0,0 +1,78 @@
+		 Alias: KeyGenerator.SP800-108-Counter
+		... ommitted ...
+
+Notice that the `Alias: lines` have extra indentation which makes it
```
I think the "lines" should not be back quoted, just the `Alias:`.

emaldona replied:
`Alias: ` - including the space, I assume that's what you intended to say.

edewata replied:
I don't think we need to include the space. We're basically just highlighting a specific text in the output.

emaldona commented:
Yes, I plan to remove it and all others there, actually the entire
directory. So, I am allowed to modify this pull request?

On Fri, May 21, 2021 at 1:02 PM Endi S. Dewata ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In samples/CapabilitiesList/README.md
> <https://github.com/dogtagpki/jss/pull/704#discussion_r637200072>:
>
> > +    sudo dnf install apache-commons-codec apache-commons-lang gcc-c++ \
> +                     java-devel jpackage-utils slf4j zlib-devel \
> +                     glassfish-jaxb-api nss-tools nss-devel cmake \
> +                     junit
>
> Are you planning to completely remove this file later, or move it
> somewhere else? If you're removing it anyway that probably should be done
> in this PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dogtagpki/jss/pull/704#discussion_r637200072>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACZPFVWHONNZGXCATRGWGNDTO235ZANCNFSM45HPIRKQ>
> .
>

edewata commented:
@emaldona Yes, the `capabilities` branch is located in your GitHub repo, so you have a full control of it. You can just remove the files/folders as another commit, then push it to the same branch again. The PR will be updated automatically.

emaldona commented:
Removed the unwanted directory via the web UI and also
modified accordingly samples/README.txt

On Fri, May 21, 2021 at 1:08 PM Elio Maldonado Batiz <
***@***.***> wrote:

> Yes, I plan to remove it and all others there, actually the entire
> directory. So, I am allowed to modify this pull request?
>
> On Fri, May 21, 2021 at 1:02 PM Endi S. Dewata ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In samples/CapabilitiesList/README.md
>> <https://github.com/dogtagpki/jss/pull/704#discussion_r637200072>:
>>
>> > +    sudo dnf install apache-commons-codec apache-commons-lang gcc-c++ \
>> +                     java-devel jpackage-utils slf4j zlib-devel \
>> +                     glassfish-jaxb-api nss-tools nss-devel cmake \
>> +                     junit
>>
>> Are you planning to completely remove this file later, or move it
>> somewhere else? If you're removing it anyway that probably should be done
>> in this PR.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/dogtagpki/jss/pull/704#discussion_r637200072>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ACZPFVWHONNZGXCATRGWGNDTO235ZANCNFSM45HPIRKQ>
>> .
>>
>

edewata commented on src/test/java/org/mozilla/jss/tests/CapabilitiesList.java:
```diff
@@ -0,0 +1,259 @@
+        /* List capabilites of the specified provider */
+        public void listCapabilities(FileWriter fw, Provider p) throws Exception {
+
+            String pName = p.getName();
```
This variable doesn't seem to be used so it can be removed.

edewata commented:
@emaldona Thanks for the update! There are still some unresolved issues above, could you take a look?

edewata commented:
@emaldona Thanks for the update! I think all code issues have been resolved, but there are still some unresolved doc issues. Please take a look at my earlier comments above. Thanks!

edewata commented:
@emaldona Thanks for the update! I think there are just 2 issues left. Would you be able to fix them as well? Thanks!

emaldona commented:
@edewata I keep scrolling up and down and haven't been able to find those 2 remaining issues. I see marks like resolved and outdated in most of them. Could you help me find those 2 issues?

edewata commented:
@emaldona I tagged you on the 2 remaining issues. They might be 'Outdated' since the source has changed, but the issues weren't completely resolved.

edewata commented:
@emaldona Great, thanks! All issues are resolved now. Do you want me to squash the commits and merge? Or do you want to squash them locally first then force push it?

emaldona commented:
Could you do the squash and merge?  Last time I messed things up.

edewata commented:
Done. Thanks!
